bin/src/server/cli.rs

```diff
@@ -0,0 +1,107 @@
+use clap::{App, Arg, SubCommand};
+
+pub struct CliArgs {
+    pub port: u16,
+    pub database_url: String,
+    pub max_open_files: Option<i32>,
+    pub bulk_load_opt: Option<bool>,
+    pub sled_compression: Option<String>,
+}
```

Comment:
A couple of options here:

1) You could get rid of `CliArgs` and just tightly couple the command line parsing logic with the construction of the proper datastore (like before.)
2) If you want to keep the parsing logic separate, this would be better represented as an enum:

```rust
pub struct CliArgs {
    pub port: u16,
    pub datastore_args: CliDatastoreArgs
}

pub enum CliDatastoreArgs {
    Memory,
    Rocksdb { path: String, max_open_files: Option<i32>, bulk_load_optimized: Option<bool>,
    Sled { path: String, compression: Option<i8> }
}
```